x/gamm/keeper/migrate.go

```diff
@@ -109,27 +109,46 @@ func (k Keeper) GetLinkedBalancerPoolID(ctx sdk.Context, concentratedPoolId uint
 	return sdk.BigEndianToUint64(balancerPoolIdBigEndian), nil
 }
 
-// OverwriteMigrationRecords sets the balancer to gamm pool migration info to the store
-// Deletes all existing records, migrationInfo in state is completely overwitten by the given migrationInfo.
-func (k Keeper) OverwriteMigrationRecords(ctx sdk.Context, migrationInfo types.MigrationRecords) {
+// deleteMigrationKeys deletes all migration records with the given prefixKey.
+func (k Keeper) deleteMigrationKeys(ctx sdk.Context, prefixKey []byte) {
 	store := ctx.KVStore(k.storeKey)
-
-	// delete all existing keys
-	prefixStore := prefix.NewStore(store, types.KeyPrefixMigrationInfoBalancerPool)
+	prefixStore := prefix.NewStore(store, prefixKey)
 	iter := prefixStore.Iterator(nil, nil)
 	defer iter.Close()
 
 	for ; iter.Valid(); iter.Next() {
 		prefixStore.Delete(iter.Key())
 	}
+}
 
-	prefixStore = prefix.NewStore(store, types.KeyPrefixMigrationInfoCLPool)
-	iter = prefixStore.Iterator(nil, nil)
-	defer iter.Close()
+// OverwriteMigrationRecordsAndRedirectDistrRecords sets the balancer to gamm pool migration info to the store and deletes all existing records
+// migrationInfo in state is completely overwitten by the given migrationInfo.
+// Additionally, the distribution record for the balancer pool is modified to redirect incentives to the new concentrated pool.
+func (k Keeper) OverwriteMigrationRecordsAndRedirectDistrRecords(ctx sdk.Context, migrationInfo types.MigrationRecords) error {
+	store := ctx.KVStore(k.storeKey)
 
-	for ; iter.Valid(); iter.Next() {
-		prefixStore.Delete(iter.Key())
+	// delete all existing keys
+	k.deleteMigrationKeys(ctx, types.KeyPrefixMigrationInfoBalancerPool)
```

Comment:
Do we need to run these deletions? Does simply running `Set` below not overwrite them?

Reply:
Discussed on call, deletion makes sense when using this method for both Replace and Update, where both methods provide the entire record to be set post deletion